### ISSUE TITLE
Failure to install from source

### ISSUE THREAD

**Issue by snayfach** [marbl/parsnp#43]
Any help would be appreciated. Here's the error I got:

```
/usr/bin/perl: symbol lookup error: /usr/common/usg/languages/perl/5.24.0/extra/lib/perl5/x86_64-linux-thread-multi/auto/List/Util/Util.so: undefined symbol: Perl_xs_handshake
Now run ./configure --prefix=/global/homes/s/snayfach && make install
Add --disable-shared to the configure line if building on Mac OS X
checking for a BSD-compatible install... /usr/bin/install -c
checking whether build environment is sane... yes
checking for a thread-safe mkdir -p... /usr/bin/mkdir -p
checking for gawk... gawk
checking whether make sets $(MAKE)... yes
checking whether the C++ compiler works... yes
checking for C++ compiler default output file name... a.out
checking for suffix of executables... 
checking whether we are cross compiling... no
checking for suffix of object files... o
checking whether we are using the GNU C++ compiler... yes
checking whether g++ accepts -g... yes
checking for style of include used by make... GNU
checking dependency style of g++... gcc3
checking whether ln -s works... yes
checking build system type... x86_64-unknown-linux-gnu
checking host system type... x86_64-unknown-linux-gnu
checking how to print strings... printf
checking for gcc... gcc
checking whether we are using the GNU C compiler... yes
checking whether gcc accepts -g... yes
checking for gcc option to accept ISO C89... none needed
checking dependency style of gcc... gcc3
checking for a sed that does not truncate output... /usr/bin/sed
checking for grep that handles long lines and -e... /usr/bin/grep
checking for egrep... /usr/bin/grep -E
checking for fgrep... /usr/bin/grep -F
checking for ld used by gcc... /global/common/genepool_sl72/usg/languages/gcc/7.1.0/x86_64-pc-linux-gnu/bin/ld
checking if the linker (/global/common/genepool_sl72/usg/languages/gcc/7.1.0/x86_64-pc-linux-gnu/bin/ld) is GNU ld... yes
checking for BSD- or MS-compatible name lister (nm)... /usr/common/usg/languages/gcc/7.1.0/bin/nm -B
checking the name lister (/usr/common/usg/languages/gcc/7.1.0/bin/nm -B) interface... BSD nm
checking the maximum length of command line arguments... 1572864
checking whether the shell understands some XSI constructs... yes
checking whether the shell understands "+="... yes
checking how to convert x86_64-unknown-linux-gnu file names to x86_64-unknown-linux-gnu format... func_convert_file_noop
checking how to convert x86_64-unknown-linux-gnu file names to toolchain format... func_convert_file_noop
checking for /global/common/genepool_sl72/usg/languages/gcc/7.1.0/x86_64-pc-linux-gnu/bin/ld option to reload object files... -r
checking for objdump... objdump
checking how to recognize dependent libraries... pass_all
checking for dlltool... no
checking how to associate runtime and link libraries... printf %s\n
checking for ar... ar
checking for archiver @FILE support... @
checking for strip... strip
checking for ranlib... ranlib
checking command to parse /usr/common/usg/languages/gcc/7.1.0/bin/nm -B output from gcc object... ok
checking for sysroot... no
checking for mt... mt
checking if mt is a manifest tool... no
checking how to run the C preprocessor... gcc -E
checking for ANSI C header files... yes
checking for sys/types.h... yes
checking for sys/stat.h... yes
checking for stdlib.h... yes
checking for string.h... yes
checking for memory.h... yes
checking for strings.h... yes
checking for inttypes.h... yes
checking for stdint.h... yes
checking for unistd.h... yes
checking for dlfcn.h... yes
checking for objdir... .libs
checking if gcc supports -fno-rtti -fno-exceptions... no
checking for gcc option to produce PIC... -fPIC -DPIC
checking if gcc PIC flag -fPIC -DPIC works... yes
checking if gcc static flag -static works... yes
checking if gcc supports -c -o file.o... yes
checking if gcc supports -c -o file.o... (cached) yes
checking whether the gcc linker (/global/common/genepool_sl72/usg/languages/gcc/7.1.0/x86_64-pc-linux-gnu/bin/ld -m elf_x86_64) supports shared libraries... yes
checking whether -lc should be explicitly linked in... no
checking dynamic linker characteristics... GNU/Linux ld.so
checking how to hardcode library paths into programs... immediate
checking whether stripping libraries is possible... yes
checking if libtool supports shared libraries... yes
checking whether to build shared libraries... yes
checking whether to build static libraries... yes
checking how to run the C++ preprocessor... g++ -E
checking for ld used by g++... /global/common/genepool_sl72/usg/languages/gcc/7.1.0/x86_64-pc-linux-gnu/bin/ld -m elf_x86_64
checking if the linker (/global/common/genepool_sl72/usg/languages/gcc/7.1.0/x86_64-pc-linux-gnu/bin/ld -m elf_x86_64) is GNU ld... yes
checking whether the g++ linker (/global/common/genepool_sl72/usg/languages/gcc/7.1.0/x86_64-pc-linux-gnu/bin/ld -m elf_x86_64) supports shared libraries... yes
checking for g++ option to produce PIC... -fPIC -DPIC
checking if g++ PIC flag -fPIC -DPIC works... yes
checking if g++ static flag -static works... yes
checking if g++ supports -c -o file.o... yes
checking if g++ supports -c -o file.o... (cached) yes
checking whether the g++ linker (/global/common/genepool_sl72/usg/languages/gcc/7.1.0/x86_64-pc-linux-gnu/bin/ld -m elf_x86_64) supports shared libraries... yes
checking dynamic linker characteristics... (cached) GNU/Linux ld.so
checking how to hardcode library paths into programs... immediate
checking for ANSI C header files... (cached) yes
checking for an ANSI C-conforming const... yes
checking for inline... inline
checking whether time.h and sys/time.h may both be included... yes
checking whether gcc needs -traditional... no
checking that generated files are newer than configure... done
configure: creating ./config.status
config.status: creating Makefile
config.status: creating libMUSCLE/Makefile
config.status: creating libMUSCLE-3.7.pc
config.status: executing depfiles commands
config.status: executing libtool commands
Making install in libMUSCLE
make[1]: Entering directory '/global/projectb/scratch/snayfach/projects/gut_mags/18_snp_trees/parsnp-1.2/muscle/libMUSCLE'
make[2]: Entering directory '/global/projectb/scratch/snayfach/projects/gut_mags/18_snp_trees/parsnp-1.2/muscle/libMUSCLE'
 /usr/bin/mkdir -p '/global/projectb/scratch/snayfach/projects/gut_mags/18_snp_trees/parsnp-1.2/muscle/lib'
 /bin/sh ../libtool   --mode=install /usr/bin/install -c   libMUSCLE-3.7.la '/global/projectb/scratch/snayfach/projects/gut_mags/18_snp_trees/parsnp-1.2/muscle/lib'
libtool: install: /usr/bin/install -c .libs/libMUSCLE-3.7.so.1.0.0 /global/projectb/scratch/snayfach/projects/gut_mags/18_snp_trees/parsnp-1.2/muscle/lib/libMUSCLE-3.7.so.1.0.0
libtool: install: (cd /global/projectb/scratch/snayfach/projects/gut_mags/18_snp_trees/parsnp-1.2/muscle/lib && { ln -s -f libMUSCLE-3.7.so.1.0.0 libMUSCLE-3.7.so.1 || { rm -f libMUSCLE-3.7.so.1 && ln -s libMUSCLE-3.7.so.1.0.0 libMUSCLE-3.7.so.1; }; })
libtool: install: (cd /global/projectb/scratch/snayfach/projects/gut_mags/18_snp_trees/parsnp-1.2/muscle/lib && { ln -s -f libMUSCLE-3.7.so.1.0.0 libMUSCLE-3.7.so || { rm -f libMUSCLE-3.7.so && ln -s libMUSCLE-3.7.so.1.0.0 libMUSCLE-3.7.so; }; })
libtool: install: /usr/bin/install -c .libs/libMUSCLE-3.7.lai /global/projectb/scratch/snayfach/projects/gut_mags/18_snp_trees/parsnp-1.2/muscle/lib/libMUSCLE-3.7.la
libtool: install: /usr/bin/install -c .libs/libMUSCLE-3.7.a /global/projectb/scratch/snayfach/projects/gut_mags/18_snp_trees/parsnp-1.2/muscle/lib/libMUSCLE-3.7.a
libtool: install: chmod 644 /global/projectb/scratch/snayfach/projects/gut_mags/18_snp_trees/parsnp-1.2/muscle/lib/libMUSCLE-3.7.a
libtool: install: ranlib /global/projectb/scratch/snayfach/projects/gut_mags/18_snp_trees/parsnp-1.2/muscle/lib/libMUSCLE-3.7.a
libtool: finish: PATH="/usr/common/usg/hpc/OFED/gnu7.1/4.0-2.0.0.1-Mellanox/bin:/usr/common/usg/languages/perl/5.24.0/extra/bin:/usr/common/usg/languages/perl/5.24.0/bin:/usr/common/usg/utility_libraries/libgd/gnu7.1/2.2.4/bin:/usr/common/usg/utility_libraries/libfreetype/gnu7.1/2.7.1/bin:/usr/common/usg/utility_libraries/libjpeg/gnu7.1/6b/bin:/usr/common/usg/utility_libraries/libpng/gnu7.1/1.6.28/bin:/usr/common/usg/languages/python/2.7-anaconda/bin:/usr/common/usg/utilities/mysql/5.7.19/bin:/usr/common/jgi/oracle_client/11.2.0.3.0/client_1/bin:/usr/common/usg/languages/java/jdk/oracle/1.8.0_31_x86_64/bin:/usr/common/usg/languages/gcc/7.1.0/bin:/usr/common/usg/bin:/usr/common/mss/bin:/usr/common/nsg/bin:/usr/syscom/nsg/bin:/usr/share/Modules/3.2.10/bin:/global/homes/s/snayfach/bin:/usr/local/bin:/usr/bin:/bin:/usr/bin/X11:/usr/games:/opt/ibutils/bin:/global/projectb/scratch/snayfach/projects/gut_mags/18_snp_trees/parsnp-1.2:/sbin" ldconfig -n /global/projectb/scratch/snayfach/projects/gut_mags/18_snp_trees/parsnp-1.2/muscle/lib
----------------------------------------------------------------------
Libraries have been installed in:
   /global/projectb/scratch/snayfach/projects/gut_mags/18_snp_trees/parsnp-1.2/muscle/lib

If you ever happen to want to link against installed libraries
in a given directory, LIBDIR, you must either use libtool, and
specify the full pathname of the library, or use the `-LLIBDIR'
flag during linking and do at least one of the following:
   - add LIBDIR to the `LD_LIBRARY_PATH' environment variable
     during execution
   - add LIBDIR to the `LD_RUN_PATH' environment variable
     during linking
   - use the `-Wl,-rpath -Wl,LIBDIR' linker flag
   - have your system administrator add LIBDIR to `/etc/ld.so.conf'

See any operating system documentation about shared libraries for
more information, such as the ld(1) and ld.so(8) manual pages.
----------------------------------------------------------------------
 /usr/bin/mkdir -p '/global/projectb/scratch/snayfach/projects/gut_mags/18_snp_trees/parsnp-1.2/muscle/bin'
  /bin/sh ../libtool   --mode=install /usr/bin/install -c muscle '/global/projectb/scratch/snayfach/projects/gut_mags/18_snp_trees/parsnp-1.2/muscle/bin'
libtool: install: /usr/bin/install -c .libs/muscle /global/projectb/scratch/snayfach/projects/gut_mags/18_snp_trees/parsnp-1.2/muscle/bin/muscle
 /usr/bin/mkdir -p '/global/projectb/scratch/snayfach/projects/gut_mags/18_snp_trees/parsnp-1.2/muscle/include/libMUSCLE-3.7/libMUSCLE'
 /usr/bin/install -c -m 644 alpha.h cluster.h clust.h clustsetdf.h clustset.h clustsetmsa.h diaglist.h distcalc.h distfunc.h dpregionlist.h dpreglist.h edgelist.h enumopts.h enums.h estring.h gapscoredimer.h gonnet.h intmath.h msadist.h msa.h muscle.h objscore.h params.h profile.h pwpath.h refine.h scorehistory.h seq.h seqvect.h textfile.h timing.h tree.h types.h unixio.h threadstorage.h '/global/projectb/scratch/snayfach/projects/gut_mags/18_snp_trees/parsnp-1.2/muscle/include/libMUSCLE-3.7/libMUSCLE'
make[2]: Leaving directory '/global/projectb/scratch/snayfach/projects/gut_mags/18_snp_trees/parsnp-1.2/muscle/libMUSCLE'
make[1]: Leaving directory '/global/projectb/scratch/snayfach/projects/gut_mags/18_snp_trees/parsnp-1.2/muscle/libMUSCLE'
make[1]: Entering directory '/global/projectb/scratch/snayfach/projects/gut_mags/18_snp_trees/parsnp-1.2/muscle'
make[2]: Entering directory '/global/projectb/scratch/snayfach/projects/gut_mags/18_snp_trees/parsnp-1.2/muscle'
make[2]: Nothing to be done for 'install-exec-am'.
 /usr/bin/mkdir -p '/global/projectb/scratch/snayfach/projects/gut_mags/18_snp_trees/parsnp-1.2/muscle/lib/pkgconfig'
 /usr/bin/install -c -m 644 libMUSCLE-3.7.pc '/global/projectb/scratch/snayfach/projects/gut_mags/18_snp_trees/parsnp-1.2/muscle/lib/pkgconfig'
make[2]: Leaving directory '/global/projectb/scratch/snayfach/projects/gut_mags/18_snp_trees/parsnp-1.2/muscle'
make[1]: Leaving directory '/global/projectb/scratch/snayfach/projects/gut_mags/18_snp_trees/parsnp-1.2/muscle'
/usr/bin/perl: symbol lookup error: /usr/common/usg/languages/perl/5.24.0/extra/lib/perl5/x86_64-linux-thread-multi/auto/List/Util/Util.so: undefined symbol: Perl_xs_handshake
/usr/bin/perl: symbol lookup error: /usr/common/usg/languages/perl/5.24.0/extra/lib/perl5/x86_64-linux-thread-multi/auto/List/Util/Util.so: undefined symbol: Perl_xs_handshake
/usr/bin/perl: symbol lookup error: /usr/common/usg/languages/perl/5.24.0/extra/lib/perl5/x86_64-linux-thread-multi/auto/List/Util/Util.so: undefined symbol: Perl_xs_handshake
build_parsnp_linux.sh: line 7: ./configure: No such file or directory
make: *** No rule to make target 'install'.  Stop.
```

**Comment by tseemann** [comment]
I can't build libMUSCLE either:
```
make
Making all in libMUSCLE
make[1]: Entering directory '/home/tseemann/tmp/parsnp-1.2/muscle/libMUSCLE'
make[1]: *** No rule to make target '../libMUSCLE/libMUSCLE-3.7.la', needed by 'muscle'.  Stop.
make[1]: Leaving directory '/home/tseemann/tmp/parsnp-1.2/muscle/libMUSCLE'
make: *** [Makefile:451: all-recursive] Error 1
```

**Comment by tseemann** [comment]
```
configure.ac:2: warning: AM_INIT_AUTOMAKE: two- and three-arguments forms are deprecated.  For more info, see:
configure.ac:2: https://www.gnu.org/software/automake/manual/automake.html#Modernize-AM_005fINIT_005fAUTOMAKE-invocation
configure.ac:3: installing './compile'
configure.ac:2: installing './install-sh'
configure.ac:2: installing './missing'
src/Makefile.am:4: warning: source file './ext/iniFile.cpp' is in a subdirectory,
src/Makefile.am:4: but option 'subdir-objects' is disabled
automake: warning: possible forward-incompatibility.
automake: At least a source file is in a subdirectory, but the 'subdir-objects'
automake: automake option hasn't been enabled.  For now, the corresponding output
automake: object file(s) will be placed in the top-level directory.  However,
automake: this behaviour will change in future Automake versions: they will
automake: unconditionally cause object files to be placed in the same subdirectory
automake: of the corresponding sources.
automake: You are advised to start using 'subdir-objects' option throughout your
automake: project, to avoid future incompatibilities.
```

**Comment by bkille** [comment]
With the current repository, the `build_parsnp_linux.sh` and `build_parsnp_osx.sh` scripts should do the trick. To do this manually, though:

In the `muscle` directory: run 
```
./autogen.sh
./configure --prefix=`pwd`
make install
```

then in the top-level directory run 
```
./autogen.sh
./configure
make install
```

**Comment by ballaneypranav** [comment]
Hey, 
`build_parsnp_linux.sh` doesn't seem to work for me. I still get the error that @tseemann has posted above. (https://github.com/marbl/parsnp/issues/43#issuecomment-412239339)

However, it works perfectly if I apply the patch Debian Med uses 
https://salsa.debian.org/med-team/libmuscle/-/blob/master/debian/patches/makefile_deps.patch
to this file here
https://github.com/marbl/parsnp/blob/master/muscle/libMUSCLE/Makefile.am#L39

Please take a look.

Regards,
Pranav

**Comment by advaitb** [comment]
@ballaneypranav Thanks for the heads up. That seems rather weird given our last muscle build fix. Just curious, are using Debian(?).  

@bkille After the last fix, I tested this on ubuntu and mint so should be compatible with debian unless some of that OS sugaring in the former distro(s) is hiding an underlying bug . I added the patch to the libMUSCLE Makefile locally now and seems to not affect build on ubuntu.
